agent/auto-config/config.go

```diff
@@ -92,9 +107,12 @@ type Config struct {
 	TLSConfigurator TLSConfigurator
 
 	// Cache is an object implementing our Cache interface. The Cache
-	// used at runtime must be able to handle Roots and Leaf Cert watches
+	// used at runtime must be able to handle Roots watches
```

Comment:
It's probably something we could do, but that should be in a followup since it does require pushing around the roots usage of the agent cache a bit more.

Likely worth revisiting when this gets uplifted into a library for `consul-dataplane`